test/extensions/filters/http/compressor/compressor_filter_integration_test.cc

```diff
@@ -493,4 +498,121 @@ TEST_P(CompressorIntegrationTest, PerRouteEnable) {
+            enabled:
+              default_value: true
+            content_type:
+              - text/html
+              - application/json
+    )EOF");
+
+  // Request to /gzip-per-route route should use the per-route compressor library.
+  // Since both are gzip in this test, we verify the structure but not actual compression
+  // difference.
+  auto response = sendRequestAndWaitForResponse(
+      Http::TestRequestHeaderMapImpl{{":method", "GET"},
+                                     {":path", "/gzip-per-route"},
+                                     {":scheme", "http"},
+                                     {":authority", "host"},
+                                     {"accept-encoding", "gzip"}},
+      0,
+      Http::TestResponseHeaderMapImpl{
+          {":status", "200"}, {"content-length", "40"}, {"content-type", "text/html"}},
+      40);
+
+  EXPECT_TRUE(response->complete());
+  EXPECT_EQ("200", response->headers().getStatusValue());
+  // Should be compressed since both main and per-route configs use gzip.
+  Http::HeaderMap::GetResult content_encoding =
+      response->headers().get(Http::CustomHeaders::get().ContentEncoding);
+  ASSERT_FALSE(content_encoding.empty());
+  EXPECT_EQ(Http::CustomHeaders::get().ContentEncodingValues.Gzip,
+            content_encoding[0]->value().getStringView());
+}
+
+// Test that per-route compressor library config creation works with various libraries.
+TEST_P(CompressorIntegrationTest, PerRouteCompressorLibraryConfigCreation) {
+  config_helper_.addConfigModifier(
+      [](envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager&
+             cm) {
+        auto* vh = cm.mutable_route_config()->mutable_virtual_hosts()->Mutable(0);
+        auto* route = vh->mutable_routes()->Mutable(0);
+        route->mutable_match()->set_path("/custom");
+
+        envoy::extensions::filters::http::compressor::v3::CompressorPerRoute per_route;
+        // Test that per-route config can be created with different compressor library.
+        auto* compressor_lib = per_route.mutable_overrides()->mutable_compressor_library();
+        compressor_lib->set_name("custom");
+        compressor_lib->mutable_typed_config()->set_type_url(
+            "type.googleapis.com/envoy.extensions.compression.gzip.compressor.v3.Gzip");
+
+        // Set some config for gzip compressor.
+        envoy::extensions::compression::gzip::compressor::v3::Gzip gzip_config;
+        gzip_config.mutable_window_bits()->set_value(12);
+        std::string serialized_config;
+        ASSERT_TRUE(gzip_config.SerializeToString(&serialized_config));
+        compressor_lib->mutable_typed_config()->set_value(serialized_config);
+
+        Any cfg_any;
+        ASSERT_TRUE(cfg_any.PackFrom(per_route));
+        route->mutable_typed_per_filter_config()->insert(
+            MapPair<std::string, Any>("envoy.filters.http.compressor", cfg_any));
+      });
+
+  initializeFilter(default_config);
```

Comment:
it seems like both the default config and the route config is set to use gzip?

I see the gzip params are specifically configured per route, but maybe we can use a different compressor just to validate the override is happening?